upgrade: Update UI texts to match final design

skazi0 opened the pull request:
The texts have been updated to match final design documents. In some
places there was a need to adapt the templates and/or styles to handle
the design changes (e.g. bold sections in text).

The final design doc is here: https://github.com/crowbar/crowbar-ui/blob/wiki-assets/upgrade67/crowbar-wizard-comp-final-text.pdf

skazi0 committed:
upgrade: Update UI texts to match final design

The texts have been updated to match final design documents. In some
places there was a need to adapt the templates and/or styles to handle
the design changes (e.g. bold sections in text).

skazi0 commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(1) S55P4 #UI: Include final ui texts](https://trello.com/c/B2HutgON/231-1-s55p4-ui-include-final-ui-texts)

jsuchome commented on assets/app/features/upgrade/i18n/en.json:
```diff
@@ -32,50 +32,50 @@
-                        "description1": "In normal mode, each node will be rebooted during the upgrade process, including the workloads running in them.",
-                        "description2": "In order to proceed in non-disruptive mode your cloud needs to have the pacemaker barclamp correctly configured and applied amd at least one free node as large or larger than any other node in your clouds compute plane.",
+                        "description-title": "IMPORTANT!",
+                        "description1": "In normal mode, the workloads running in each node will be live-migrated to another node. Note that you will lose connection to your VMs when upgrading the control node(s).",
```
We actually still have not decided what the disruptive mode will do. I hope this is correct, but it may need another change later.

skazi0 replied:
We will need to follow up with some changes once we get this clarified.

jsuchome commented on assets/app/features/upgrade/i18n/en.json:
```diff
@@ -32,50 +32,50 @@
                     "nondisruptive": {
                         "title": "Your cloud is now prepared to begin the upgrade process in non-disruptive mode.",
-                        "description1": "In non-disruptive mode all workloads should continue running but we strongly recommend picking your nose and wiggeling your finger at it for a while first.",
+                        "description1": "In non-disruptive mode all workloads should continue running and remain connected.",
```
*should* or *will*? I'm also not sure about *connected* - what would you think about *accessible from outside* or something like that?

skazi0 replied:
I wouldn't go for _will_ here as there's always some possibility of problems with workloads during migration. I don't like "remain connected" too. Sounds like they are connected to one another.

jsuchome commented on assets/app/features/upgrade/i18n/en.json:
```diff
@@ -32,50 +32,50 @@
-                "title": "Upgrading to SUSE OpenStack Cloud 7 requires software repositories for SLES12 SP2 and SUSE OpenStack Cloud 7.",
-                "description": "Update the URLs of the following software repositories and click 'Check' to verify the repositories are available. When all repositories have been verified click on 'Next' to proceed.",
+                "title": "Upgrading to SUSE OpenStack Cloud 7 requires updated software repositories for the Administration Server",
+                "description": "Prepare the software repositories locally or via SMT and update the URLs in the Administration Server configuration correspondingly. Click \"Check\" to verify all repositories are available. When all repositories have been verified click on \"Next\" to proceed.",
```
It might be good to add a sentence in a sense *Refer to **Software Repository Setup** chapter of the deployment guide.*

jsuchome commented on assets/app/features/upgrade/i18n/en.json:
```diff
@@ -32,50 +32,50 @@
             "database-configuration": {
-                "name": "",
-                "title": "Enter Credentials and/or Information to Create or Connect to an OpenStack PostgreSQL Server for the OpenStack Database Migration.",
+                "title": "Enter Credentials and/or Information to create or connect to a PostgreSQL database for the Crowbar database",
```
That sentence looks weird. What about *...to a PostgreSQL database that should serve as the Crowbar database* ?

jsuchome commented on assets/app/features/upgrade/i18n/en.json:
```diff
@@ -87,25 +87,26 @@
             "openstack-services": {
-                "title": "Stop OpenStack Services and Create Backup",
-                "description": "When ready, click on the button below to stop OpenStack services and create a backup of the OpenStack database.",
+                "title": "Stop all OpenStack services and back up the OpenStack database",
```
No, we are not stopping *all* services, only most of them.

jsuchome commented on assets/app/features/upgrade/i18n/en.json:
```diff
@@ -87,25 +87,26 @@
-                "title": "Stop OpenStack Services and Create Backup",
-                "description": "When ready, click on the button below to stop OpenStack services and create a backup of the OpenStack database.",
+                "title": "Stop all OpenStack services and back up the OpenStack database",
+                "description": "It is imperative that during and after stopping the OpenStack services no changs are made to your cloud. Once created, the backup can be found in /var/lib/crowbar/backup on Administration Server.",
```
We should say at some point that the *OpenStack API won't be available until the upgrade is finished*.

Itxaka commented:
I think @jsuchome comments make sense but shouldn that be raised against @kwwii instead to have changes to the design doc?

jsuchome commented:
I know I've read those texts before already, and not just me - but you always overlook something. I think it would not make sense to make another round of design -> implementation, I think it should be fine if @kwwii agrees with these changes in this PR.

skazi0 commented:
Thanks for the comments. The base of this PR is the document from @kwwii. I agree that there are still some things to improve. It would be good if we can get @kwwii involved again.

One thing I already changed wrt the document is on the landing page: I added "mandatory" to the sentence about all checks being successful. More changes to come.

skazi0 committed:
upgrade: Improve some UI texts after review

Follow up after another review round.
